src/main/webapp/WEB-INF/view/chat.jsp

```diff
@@ -57,10 +66,50 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
       <ul>
     <%
       for (Message message : messages) {
+      /*%><div class = "profilenames" style= "color: green">
+        <%*/
         String author = UserStore.getInstance()
           .getUser(message.getAuthorId()).getName();
-    %>
-      <li><strong><%= author %>:</strong> <%= message.getContent() %></li>
+          %>
+      <li><div id="profileName"><%= author %>:</div>
+        <div id="commonInterests">
+          <%
+          
+          String result = "Common interests: ";
+          String currentUser = (String)request.getSession().getAttribute("user");
+
+          if (currentUser != null){
+            User currUser = UserStore.getInstance().getUser(currentUser);
+            User otherUser = UserStore.getInstance().getUser(author);
+            UserProfile currProf = UserProfileStore.getInstance().getUserProfile(currUser.getId());
+            UserProfile otherProf = UserProfileStore.getInstance().getUserProfile(otherUser.getId());
+            HashMap<String,String> currentInterests = (HashMap<String,String>)currProf.getInterests();
+            HashMap<String,String> otherInterests = (HashMap<String,String>)otherProf.getInterests();
+
+            for(String interest: currentInterests.keySet()){
+              if(otherInterests.containsKey(interest)){
+                String currInterest = currentInterests.get(interest);
+                String[] currSplit = currInterest.split(", ");
+
+                String otherInterest = otherInterests.get(interest);
+                String[] otherSplit = otherInterest.split(", ");
+
+                for(int i = 0; i < currSplit.length; i++)
+                  for(int j = 0; j < otherSplit.length; j++)
+                      if(currSplit[i].equals(otherSplit[j]))
+                        result += " " + currSplit[i] + ",";
+                      
+
+              }
+            }
+            if(result.equals("Common interests: "))
+              result = "No interests in common.";
+            else
+              result = result.substring(0, result.length()-1);
```

Comment:
This is a nice refinement! I would add a comment explaining that it removes the trailing comma from the last list item.

An alternative would be to build up a list of strings instead of the fully formatted string directly, then check if the list is empty and join them into the display string at the end. That may be a little nicer than dealing directly with Java's immutable Strings everywhere.

I think that then there would also be fancy ways to generate the list of matching interests using Java streams and filters--let me know if you're interested in discussing what those would look like!